### PR TITLE
MmapShMem Fix

### PR DESCRIPTION
**Pull request by tokatoka**

### CODE REVIEW

File: libafl/src/bolts/shmem.rs [hunk 1]
```diff
@@ -610,14 +613,14 @@ pub mod unix_shmem {
 
         impl MmapShMem {
             /// Create a new [`MmapShMem`]
-            pub fn new(map_size: usize, shmem_ctr: usize) -> Result<Self, Error> {
+            pub fn new(map_size: usize) -> Result<Self, Error> {
                 unsafe {
-                    let mut filename_path = [0_u8; MAX_MMAP_FILENAME_LEN];
+                    let mut filename_path = [0_u8; MAX_MMAP_FILENAME_LEN]; // This is the filename string used to search for shmem.
+                    let random_id = StdRand::with_seed(read_time_counter()).next_u32() as i32;
```
Comment (domenukk) [inline libafl/src/bolts/shmem.rs:619]: I'd store an initialized rand in the provider

**Comment by tokatoka** [comment]
I used u32 (not u64) because unix_shmem_server.rs assumes ShMemId is a i32 number (and I didn't want to break codes in shmem server)

**Comment by tokatoka** [comment]
this does not work unix shmem server :<
it assumes ShMemId is fd

File: libafl/src/bolts/shmem.rs [hunk 2]
```diff
@@ -53,7 +53,7 @@ pub type StdShMemService = ShMemService<MmapShMemProvider>;
     unix,
     not(any(target_os = "android", target_vendor = "apple"))
 ))]
-pub type StdShMemProvider = UnixShMemProvider;
+pub type StdShMemProvider = MmapShMemProvider;
```
Comment (domenukk) [inline libafl/src/bolts/shmem.rs:56]: Why change the Standard?

File: libafl/src/bolts/shmem.rs [hunk 3]
```diff
@@ -656,23 +660,45 @@ pub mod unix_shmem {
...
+                        id,
                     })
                 }
             }
 
+            #[allow(clippy::cast_sign_loss)]
             fn shmem_from_id_and_size(id: ShMemId, map_size: usize) -> Result<Self, Error> {
                 unsafe {
-                    let shm_fd: i32 = id.to_string().parse().unwrap();
+                    let id: i32 = id.into();
```
Comment (domenukk) [inline libafl/src/bolts/shmem.rs:682]: why i32 and not u32?

**Comment by tokatoka** [comment]
What this PR is trying to do is wrong.. This ShmId needs to be a file descriptor for unix shmem server. Closing this.

**Comment by domenukk** [comment]
I'm not sure that's always the case? Maybe we need to change the server instead? 😀

**Pull request closed without merging**